parachains/runtimes/bridge-hubs/bridge-hub-rococo/tests/tests.rs

```diff
@@ -129,6 +151,341 @@ mod bridge_hub_rococo_tests {
 		}),
 		bridge_hub_rococo_config::DEFAULT_XCM_LANE_TO_BRIDGE_HUB_WOCOCO
 	);
+
+	bridge_hub_test_utils::include_relayed_incoming_message_works!(
+		Runtime,
+		XcmConfig,
+		ParachainSystem,
+		BridgeGrandpaWococoInstance,
+		BridgeParachainWococoInstance,
+		WithBridgeHubWococoMessagesInstance,
+		WithBridgeHubWococoMessageBridge,
+		bridge_hub_test_utils::CollatorSessionKeys::new(
+			AccountId::from(ALICE),
+			AccountId::from(ALICE),
+			SessionKeys { aura: AuraId::from(sp_core::sr25519::Public::from_raw(ALICE)) }
+		),
+		bp_bridge_hub_rococo::BRIDGE_HUB_ROCOCO_PARACHAIN_ID,
+		bp_bridge_hub_wococo::BRIDGE_HUB_WOCOCO_PARACHAIN_ID,
+		1000,
+		LaneId([0, 0, 0, 0]),
+		Rococo,
+	);
+
+	#[test]
+	fn receive_single_message_proof() {
+		bridge_hub_test_utils::test_cases::receive_single_message_proof::<
+			Runtime,
+			XcmConfig,
+			ParachainSystem,
+			BridgeGrandpaWococoInstance,
+			BridgeParachainWococoInstance,
+			WithBridgeHubWococoMessagesInstance,
+			WithBridgeHubWococoMessageBridge,
+		>(
+			bridge_hub_test_utils::CollatorSessionKeys::new(
+				AccountId::from(ALICE),
+				AccountId::from(ALICE),
+				SessionKeys { aura: AuraId::from(sp_core::sr25519::Public::from_raw(ALICE)) },
+			),
+			bp_bridge_hub_rococo::BRIDGE_HUB_ROCOCO_PARACHAIN_ID,
+			1000,
+		);
+	}
+
+	pub fn construct_extrinsic(
+		sender: sp_core::sr25519::Pair,
+		call: RuntimeCall,
+	) -> UncheckedExtrinsic {
+		let extra: SignedExtra = (
+			frame_system::CheckNonZeroSender::<Runtime>::new(),
+			frame_system::CheckSpecVersion::<Runtime>::new(),
+			frame_system::CheckTxVersion::<Runtime>::new(),
+			frame_system::CheckGenesis::<Runtime>::new(),
+			frame_system::CheckEra::<Runtime>::from(Era::immortal()),
+			frame_system::CheckNonce::<Runtime>::from(0),
+			frame_system::CheckWeight::<Runtime>::new(),
+			pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::from(0),
+			BridgeRejectObsoleteHeadersAndMessages {},
+			(
+				BridgeRefundBridgeHubRococoMessages::default(),
+				BridgeRefundBridgeHubWococoMessages::default(),
+			),
+		);
+
+		let payload = SignedPayload::new(call.clone(), extra.clone()).unwrap();
+		let signature = payload.using_encoded(|e| sender.sign(e));
+
+		UncheckedExtrinsic::new_signed(
+			call,
+			AccountId32::from(sender.public()).into(),
+			Signature::Sr25519(signature.clone()),
+			extra,
+		)
+	}
+
+	#[test]
+	pub fn complex_relay_extrinsic_works() {
+		use asset_test_utils::{mock_open_hrmp_channel, ExtBuilder};
+
+		test(
+			bridge_hub_test_utils::CollatorSessionKeys::new(
+				AccountId::from(ALICE),
+				AccountId::from(ALICE),
+				SessionKeys { aura: AuraId::from(sp_core::sr25519::Public::from_raw(ALICE)) },
+			),
+			bp_bridge_hub_rococo::BRIDGE_HUB_ROCOCO_PARACHAIN_ID,
+			bp_bridge_hub_wococo::BRIDGE_HUB_WOCOCO_PARACHAIN_ID,
+			1000,
+			LaneId([0, 0, 0, 1]),
```

Comment:
@svyatonik I spent some time debugging the relayer refund extension only to find I was using wrong `LaneId`. The potentially concerning part is that besides the relayer not getting refund, everything else "worked" even when using `LaneId([0, 0, 0, 0])` here - unless I'm missing smth, relayer can just submit "valid" proofs to "wrong" lanes and they're accepted.
Can you also have a look and see if we're misconfiguring something or what?

Reply:
We have the [similar issue](https://github.com/paritytech/parity-bridges-common/issues/1759) to add the `lane <> sender` check. But now it isn't super critical, because we have the single lane/bridge. Sorry for the wasted time - I hope to make pallets configuration easier after initial deployment